Feature/data length

leoTiez opened the pull request.

leoTiez committed:
Update: added parameter for data length for easy parameter tests

leoTiez committed:
Update: Added the train and test data length as parameter

Train and test data length are now passed as parameter to the
load data function

adrianc-a commented on load_data.py:
```diff
@@ -97,4 +97,7 @@ def data_loader(dataset, file_name, dtype, use_grayscale=False):
+    if data_length is None:
+        return data, labels
+    else:
+        return data[:data_length], labels[:data_length]
```
let's shuffle this and i'll approve it.

leoTiez replied:
This is not a good idea, since we want to use it for the comparison of the hyperparameters. They should all be tested on the same subset

leoTiez replied:
No you are right, this is important for the distribution of the classes

leoTiez committed:
Bugfix: Changed learning rate value

Since problems occured with having nan as costs the learning
rate has was decreased

leoTiez committed:
Update: Changed default parameter for min difference accuracy

To provide a more reasonable update behavior

leoTiez committed:
Update: Shuffle data first to uniformly distribute samples

leoTiez committed:
Update: Changed min diff to a more reasonable default value
